README.md

```diff
@@ -122,6 +122,70 @@ If you are using the gui, we recommend that you activate advanced mode. Also you
 
 After generating the build configuration, just open the Sourcetrail.sln file that was generated by CMake and build the Sourcetrail project.
 
+##### If using clang-cl.exe and Ninja (such as via CLion)
+
+Here's what I ended up using for Clang/Ninja builds via CLion:
+
+* [Build Tools for Visual Studio 2019](https://aka.ms/buildtools) (as an alternative to installing Visual Studio)
+* [Windows 10 SDK](https://developer.microsoft.com/en-us/windows/downloads/windows-10-sdk/) for released versions of Windows. If running a Windows Insider preview, you'll need the [Preview SDK](https://insider.windows.com/en-us/for-developers/) instead.
+* Boost 1.72.0 compiled [following Windows instructions](https://www.boost.org/doc/libs/1_72_0/more/getting_started/windows.html) from within [a Build Tools command prompt](https://docs.microsoft.com/en-us/cpp/build/building-on-the-command-line?view=vs-2019#developer_command_file_locations) (or you can download a [pre-compiled set of binaries](https://bintray.com/boostorg/release/boost-binaries)).
```

Comment:
We currently use Boost 1.67, which is the version used in certain Linux distributions. We currently don't plan to update this to a higher version, unless there is a good reason.

Reply:
1.67 is two days shy of being two years old. While the changelogs don’t reveal anything especially interesting since then, they do mention adding private visibility for smaller shared libraries, adding more C++17 feature compatibility and so on.

Boost, like Qt, might be the type of library where we need CI runs with more than one version: the minimum we support, and then the maximum we support. Similar to Mac, Windows and Linux compatibility, we could define platform to include these different variants. The easiest way to do that might be to define a set of Linux OS releases to support and test compiling and linking on those OSes.

Reply:
It wouldn’t be perfect (kernel, etc.) but using Docker we can mimic the filesystem and dependencies shipped by various operating systems, and list those OSes as platforms we support. My preference, though it is just a preference, would be for us to always support the latest version, and introduce at most what I would call temporary hacks or workarounds to support previous versions. Boost is a stable library, it’s easy to support multiple versions. Same with Qt if you’re not trying to use the latest and greatest. Clang on the other hand seems incredibly unstable, so we probably can’t support back farther than one, maybe two, releases, and new releases seem to occur every 6 months. Java on the other hand seems relatively stable but again you can only use features from the oldest version of Java you’d want to support...

Reply:
I generally support updating software to the newest version. However, regarding Boost there is really not much of a benefit for us in doing so. And testing with multiple Boost versions is not a problem worth solving in my opinion.

Our officially supported version is `1.67` and everything in our repository should consistently reflect this choice. If you use a different version that also works, that is your personal choice. When there is a good reason to update, then we will move forward.